dotCMS/src/integration-test/java/com/dotcms/publishing/PublisherAPITest.java

```diff
@@ -319,4 +326,303 @@ private FileFilter getNoBundleXMLFileFilter() {
         return file -> !file.getAbsolutePath().endsWith("bundle.xml");
     }
 
+    /**
+     * Method to test: {@link PublisherAPI#addFilterDescriptor(FilterDescriptor)}
+     * Given Scenario: Create a new FilterDescriptor and add it to the FilterDescriptorMap
+     * ExpectedResult: the filterDescriptor is added successfully to the map
+     *
+     */
+    @Test
+    public void test_addFilter_success(){
+        publisherAPI.getFilterDescriptorMap().clear();
+
+        final Map<String,Object> filtersMap =
+                ImmutableMap.of("dependencies",true,"relationships",true,"excludeClasses","Host,Workflow");
+        final FilterDescriptor filterDescriptor =
+                new FilterDescriptor("filterTestAPI.yml","Filter Test Title",filtersMap,false,"Reviewer,dotcms.org.2789");
+
+        publisherAPI.addFilterDescriptor(filterDescriptor);
+
+        final Map<String,FilterDescriptor> filterDescriptorMap = APILocator.getPublisherAPI().getFilterDescriptorMap();
+        Assert.assertFalse(filterDescriptorMap.isEmpty());
+        Assert.assertTrue(filterDescriptorMap.containsKey(filterDescriptor.getKey()));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Assert.assertTrue' due to existing static import 'org.junit.Assert.assertTrue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)